--- packages/lexical-code-prism/src/CodeHighlighterPrism.ts
+++ packages/lexical-code-prism/src/CodeHighlighterPrism.ts
@@ -143,18 +143,18 @@
   const language = node.getLanguage() || tokenizer.defaultLanguage;
   if (language) {
     if (isCodeLanguageLoaded(language)) {
-      if (!node.getIsSyntaxHighlightSupported()) {
-        node.setIsSyntaxHighlightSupported(true);
-      }
+      if (node.getIsSyntaxHighlightSupported() !== true) {
+        node.setIsSyntaxHighlightSupported(true);
+      }
     } else {
-      if (node.getIsSyntaxHighlightSupported()) {
-        node.setIsSyntaxHighlightSupported(false);
-      }
+      if (node.getIsSyntaxHighlightSupported() !== false) {
+        node.setIsSyntaxHighlightSupported(false);
+      }
       loadCodeLanguage(language, editor, nodeKey);
       return;
     }
-  } else if (node.getIsSyntaxHighlightSupported()) {
-    node.setIsSyntaxHighlightSupported(false);
+  } else if (node.getIsSyntaxHighlightSupported() !== false) {
+    node.setIsSyntaxHighlightSupported(false);
   }
 
   if (nodesCurrentlyHighlighting.has(nodeKey)) {
--- packages/lexical-code-shiki/src/CodeHighlighterShiki.ts
+++ packages/lexical-code-shiki/src/CodeHighlighterShiki.ts
@@ -170,18 +170,18 @@
   // dynamic import of languages
   if (language) {
     if (isCodeLanguageLoaded(language)) {
-      if (!node.getIsSyntaxHighlightSupported()) {
-        node.setIsSyntaxHighlightSupported(true);
-      }
+      if (node.getIsSyntaxHighlightSupported() !== true) {
+        node.setIsSyntaxHighlightSupported(true);
+      }
     } else {
-      if (node.getIsSyntaxHighlightSupported()) {
-        node.setIsSyntaxHighlightSupported(false);
-      }
+      if (node.getIsSyntaxHighlightSupported() !== false) {
+        node.setIsSyntaxHighlightSupported(false);
+      }
       loadCodeLanguage(language, editor, nodeKey);
       inFlight = true;
     }
-  } else if (node.getIsSyntaxHighlightSupported()) {
-    node.setIsSyntaxHighlightSupported(false);
+  } else if (node.getIsSyntaxHighlightSupported() !== false) {
+    node.setIsSyntaxHighlightSupported(false);
   }
 
   if (inFlight) {
--- packages/lexical-code-shiki/src/FacadeShiki.ts
+++ packages/lexical-code-shiki/src/FacadeShiki.ts
@@ -77,7 +77,7 @@
             if (
               $isCodeNode(codeNode) &&
               codeNode.getLanguage() === language &&
-              !codeNode.getIsSyntaxHighlightSupported()
+              codeNode.getIsSyntaxHighlightSupported() !== true
             ) {
               codeNode.setIsSyntaxHighlightSupported(true);
             }
